apps/server/src/infra/column-client/column-client.adapter.spec.ts

```diff
@@ -0,0 +1,83 @@
+import { faker } from '@faker-js/faker';
+import { createMock } from '@golevelup/ts-jest';
+import { Test, TestingModule } from '@nestjs/testing';
+import { ColumnClientAdapter } from './column-client.adapter';
+import { BoardColumnApi, CreateCardBodyParamsRequiredEmptyElements } from './generated';
+
+describe(ColumnClientAdapter.name, () => {
+	let module: TestingModule;
+	let sut: ColumnClientAdapter;
+
+	const columnApiMock = createMock<BoardColumnApi>();
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				ColumnClientAdapter,
+				{
+					provide: BoardColumnApi,
+					useValue: columnApiMock,
+				},
+			],
+		}).compile();
+		sut = module.get(ColumnClientAdapter);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	it('should be defined', () => {
+		expect(sut).toBeDefined();
+	});
+
+	describe('createCard', () => {
+		describe('when creating a card', () => {
+			const setup = () => {
+				const columnId = faker.string.uuid();
+				const cardParams = {
+					requiredEmptyElements: [CreateCardBodyParamsRequiredEmptyElements.RICH_TEXT],
+				};
+
+				return {
+					columnId,
+					cardParams,
+				};
+			};
+
+			it('should call boardColumnApi.ColumnControllerCreateCard', async () => {
+				const { columnId, cardParams } = setup();
+
+				await sut.createCard(columnId, cardParams);
+
+				expect(columnApiMock.columnControllerCreateCard).toHaveBeenCalledWith(columnId, cardParams);
+			});
+		});

```

Comment:
Do we want to extend the test and check the response?

```suggestion
		describe('when creating a card', () => {
			const setup = () => {
				const columnId = faker.string.uuid();
				const cardParams = {
					requiredEmptyElements: [CreateCardBodyParamsRequiredEmptyElements.RICH_TEXT],
				};

				columnApiMock.columnControllerCreateCard.mockResolvedValue(axiosResponseFactory.build({ data: columnId }));

				return {
					columnId,
					cardParams,
				};
			};

			it('should call boardColumnApi.ColumnControllerCreateCard', async () => {
				const { columnId, cardParams } = setup();

				const response = await sut.createCard(columnId, cardParams);

				expect(response).toEqual(columnId);
				expect(columnApiMock.columnControllerCreateCard).toHaveBeenCalledWith(columnId, cardParams);
			});
		});
```